File: src/basics/patterns/Pattern7.java
package src.basics.patterns;

public class Pattern7 {
    /*
    Pattern 7:
    https://practice.geeksforgeeks.org/problems/triangle-pattern-1661492263/1

        *
       ***
      *****
     *******
    *********

     */

    static void printTriangle(int n) {
        // code here
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < (n - 1 - i); j++) {
                System.out.print(" ");
            }
            for (int k = 0; k < (i * 2 + 1); k++) {
                System.out.print("*");
            }
            // this part can be removed without affecting result
//            for (int l = 0; l < (n - 1 - i); l++) {
//                System.out.print(" ");
//            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        printTriangle(8);
    }
}

File: basics/patterns/pattern7.py
"""
    Pattern 7:
    https://practice.geeksforgeeks.org/problems/triangle-pattern-1661492263/1

        *
       ***
      *****
     *******
    *********
"""


def printTriangle(n):
    for i in range(n):
        for j in range(n-i-1):
            print(end=" ")

        for j in range(2*i+1):
            print("*", end="")

        for j in range(n-i-1):
            print(end=" ")

        print()


if __name__ == "__main__":
    printTriangle(5)
